packages/firestore/test/unit/specs/limbo_spec.test.ts

```diff
@@ -434,4 +434,59 @@ describeSpec('Limbo Documents:', [], () => {
         .expectEvents(query, { removed: [docC] });
     }
   );
+
+  specTest('Limbo documents stay consistent between views', [], () => {
+    // This tests verifies that a document is consistent between views, even
+    // if the document is only in Limbo in one of the.
+    const originalQuery = Query.atPath(path('collection')).addFilter(
+      filter('matches', '==', true)
+    );
+    const filteredQuery = Query.atPath(path('collection'))
+      .addFilter(filter('matches', '==', true))
+      .addFilter(filter('key', '>=', 'a'));
+
+    const docA = doc('collection/a', 1000, { key: 'a', matches: true });
```

Comment:
FWIW These names read almost backwards to me. Without looking too hard at the contents of the docs, it seems to me as if docACommitted would be the state after we've committed. As it stands it's docACommitted while we're waiting for the commit and then docA after.

docADirty might be a better name for docA with hasCommittedMutations.

Reply:
Those names work for me, and they are a little clearer (at least for everyone that remembers Jonny making up songs about the "dirty bit"). I changed the names to docA, docADirty and docBDirty.